utils/src/main/java/org/corfudb/utils/lock/LockClient.java

```diff
@@ -96,12 +98,75 @@ public LockClient(UUID clientId, CorfuRuntime corfuRuntime) throws NoSuchMethodE
             t.setDaemon(true);
             return t;
         });
-
+        this.lockConfig = lockConfig;
         this.clientId = clientId;
-        this.lockStore = new LockStore(corfuRuntime, clientId);
+        this.lockStore = new LockStore(corfuRuntime, clientId, lockConfig);
         this.clientContext = new ClientContext(clientId, lockStore, taskScheduler, lockListenerExecutor);
     }
 
+    /**
+     * Application registers interest for a lock [lockgroup, lockname]. The <class>Lock</class> will then
+     * make periodic attempts to acquire lock. Lock is acquired when the <class>Lock</class> is able to write
+     * a lease record in a common table that is being written to/read by all the registered <class>Lock</class>
+     * instances. Once acquired, the lease for the lock needs to be renewed periodically or else it will be acquired by
+     * another contending <class>Lock</class> instance. The application is notified if a lock is lost.
+     *
+     * @param lockListener
+     */
+    public void registerInterest(LockListener lockListener) {
+        registerInterest(lockConfig.getLockGroup(), lockConfig.getLockName(),
+                lockListener);
+    }
+
+    @VisibleForTesting
+    public void deregisterInterest() {
+        LockId lockId = LockDataTypes.LockId.newBuilder()
+                .setLockGroup(lockConfig.getLockGroup())
+                .setLockName(lockConfig.getLockName())
+                .build();
+        if (!locks.containsKey(lockId)){
+            throw new IllegalStateException("Can only deregister if the known " +
+                    "lock is present in the map.");
+        }
+        Lock lock = locks.get(lockId);
+
+        lock.input(LockEvent.LEASE_EXPIRED);
+    }
+
+    @VisibleForTesting
+    public void resumeInterest() {
+        LockId lockId = LockDataTypes.LockId.newBuilder()
+                .setLockGroup(lockConfig.getLockGroup())
+                .setLockName(lockConfig.getLockName())
+                .build();
+        if (!locks.containsKey(lockId)){
+            throw new IllegalStateException("Can only resume interest if the known " +
+                    "lock is present in the map.");
+        }
+
+        Lock lock = locks.get(lockId);
+
+        monitorLocks();
+
+        lock.input(LockEvent.LEASE_REVOKED);
```

Comment:
why do we revoke the lease on resumeInterest?  Also we should add the same validation here.

Reply:
It has the same logic as registerInterest.

Reply:
@pankti-m that's how the register/resume interest in lock works. Basically we call the revoke lease in that case to resume the interest. I added validation.